app/plugin.go

```diff
@@ -567,3 +622,54 @@ func (a *App) notifyPluginEnabled(manifest *model.Manifest) error {
 
 	return nil
 }
+
+func (a *App) writeFile(filename string, key io.Reader) *model.AppError {
+	data, err := ioutil.ReadAll(key)
+	if err != nil {
+		return model.NewAppError("writeFile", "app.plugin.write_file.read.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	err = a.Srv.configStore.SetFile(filename, data)
```

Comment:
Should we store the keys within a folder in the file store?

Reply:
configStore is not necessarily a file store it might be backed by a DB. We basically have a key value store for all purposes. That is why extension is needed(to avoid collisions with other types of files).